### DISCUSSION TITLE
Feature Request: Provide CanExecute() method with a new attribute [CanExecuteDependsOnProperty(nameof(IsBusy))]

### DISCUSSION THREAD

**Discussion by suugbut** [CommunityToolkit/Maui#2184] [New Feature Discussions]
In my opinion `ObservableProperty` (as a publisher) should not know anything about `ICommand` (as a subscriber). So the following
```cs
public partial class PersonViewModel : ObservableObject
{
    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(SaveCommand))]
    private boo isBusy;

    [RelayCommand(CanExecute = nameof(CanSave))]
    private void Save() { }
    private bool CanSave() => !IsBusy;
}
```
is more elegant if it can be written as follows.
```cs
public partial class PersonViewModel : ObservableObject
{
    [ObservableProperty]
    private bool isBusy;

    [RelayCommand(CanExecute = nameof(CanSave))]
    private void Save() { }
    
    [CanExecuteDependsOnProperty(nameof(IsBusy))] // <======= the requested attribute.
    private bool CanSave() => !IsBusy;
}
```

With `CanExecuteDependsOnProperty` we can also make `CanSave()`, for example, depend on properties defined in the base class.
```cs
public partial class BaseViewModel : ObservableObject
{
    [ObservableProperty]
    private bool isBusy;
}
public partial class PersonViewModel : BaseViewModel
{
    [RelayCommand(CanExecute = nameof(CanSave))]
    private void Save() { }
    
    [CanExecuteDependsOnProperty(nameof(IsBusy))] // <======= the requested attribute.
    private bool CanSave() => !IsBusy;
}
```
Any comments are welcome.

**Comment by bijington** [comment]
At first glance the detail looks good but I'm afraid you have the wrong repository. This is for the .NET MAUI Community Toolkit but you want the [.NET Community Toolkit](https://github.com/CommunityToolkit/dotnet). I don't have the power to transfer issue so can you please open it there and I will close one down. Feel free to refer to this in the new issue to help others if they stumble across this.
